godot-core/src/builtin/mod.rs

```diff
@@ -128,3 +128,205 @@ pub(crate) fn u8_to_bool(u: u8) -> bool {
         _ => panic!("Invalid boolean value {u}"),
     }
 }
+
+/// Clippy often complains if you do `f as f64` when `f` is already an `f64`.
+/// This trait exists to make it easy to convert between the different reals
+/// and floats without a lot of allowing clippy lints for your code.
+pub trait RealConv {
+    /// Cast this [`real`] to an [`f32`] using `as`.
+    // Clippy complains that this is an `as_*` function but it takes a `self`
+    // however, since this uses `as` internally it makes much more sense for
+    // it to be named `as_f32` rather than `to_f32`.
+    #[allow(clippy::wrong_self_convention)]
+    fn as_f32(self) -> f32;
+
+    /// Cast this [`real`] to an [`f64`] using `as`.
+    // Clippy complains that this is an `as_*` function but it takes a `self`
+    // however, since this uses `as` internally it makes much more sense for
+    // it to be named `as_f64` rather than `to_f64`.
+    #[allow(clippy::wrong_self_convention)]
+    fn as_f64(self) -> f64;
+
+    /// Cast an [`f32`] to a [`real`] using `as`.
+    fn from_f32(f: f32) -> Self;
+
+    /// Cast an [`f64`] to a [`real`] using `as`.
+    fn from_f64(f: f64) -> Self;
+}
+
+#[cfg(not(feature = "precision-double"))]
+mod real_mod {
+    //! Definitions for single-precision `real`.
+
+    /// Floating point type used for many structs and functions in Godot.
+    ///
+    /// This is not the `float` type in gdscript, that type is always 64-bits.
+    /// Rather, many structs in Godot may use either 32-bit or 64-bit floats
+    /// such as [`Vector2`](super::Vector2). To convert between [`real`] and
+    /// [`f32`] or [`f64`] see [`RealConv`](super::RealConv).
+    ///
+    /// See also the [Godot docs on float](https://docs.godotengine.org/en/stable/classes/class_float.html).
+    ///
+    /// _Godot equivalent: `real_t`_
+    // As this is a scalar value, we will use a non-standard type name.
+    #[allow(non_camel_case_types)]
+    pub type real = f32;
+
+    impl super::RealConv for real {
+        #[inline]
+        fn as_f32(self) -> f32 {
+            self
+        }
+
+        #[inline]
+        fn as_f64(self) -> f64 {
+            self as f64
+        }
+
+        #[inline]
+        fn from_f32(f: f32) -> Self {
+            f
+        }
+
+        #[inline]
+        fn from_f64(f: f64) -> Self {
+            f as f32
+        }
+    }
+
+    pub use std::f32::consts;
+
+    /// A glam [`Vec2`](glam::Vec2) with floating-point format compatible
+    /// with [`real`].
```

Comment:
Could you use a single line for these comments?

In general you use very short line widths, while screens are super wide 😉 probably a matter of taste, but you could easily use 1.5x your current width in my opinion.

Reply:
That's fair, it'd be nice if there was a standard line width in the project tho

Reply:
The problem is, as long as there is no tooling to enforce it, such a convention will just be ignored in every PR, and I can spend the rest of my free time bikeshedding line breaks 😉 

rustfmt nightly has [`max_width`](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#max_width), which may be a start -- but it seems like it only breaks longer lines, and doesn't extend shorter ones -- which would make it rather useless.

Reply:
I can at least say that i would follow it if there was one